src/csvcubed/writers/qbwriter.py

```diff
@@ -81,6 +81,29 @@ class QbWriter(WriterBase):
     def csv_metadata_file_name(self) -> str:
         return f"{self.csv_file_name}-metadata.json"
 
+    @property
+    def is_cube_in_pivoted_shape(self) -> bool:
+        obs_val_columns = get_columns_of_dsd_type(self.cube, QbObservationValue)
+
+        all_pivoted = True
+        all_standard_shape = True
+        for obs_val_col in obs_val_columns:
+            all_pivoted = (
+                all_pivoted
+                and obs_val_col.structural_definition.is_pivoted_shape_observation
+            )
+            all_standard_shape = (
+                all_standard_shape
+                and not obs_val_col.structural_definition.is_pivoted_shape_observation
+            )
+
+        if all_pivoted:
+            return True
+        elif all_standard_shape:
+            return False
+        else:
+            raise Exception("The cube cannot be in both standard and pivoted shape")
```

Comment:
Can you use [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError) here?